packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -756,6 +756,41 @@ mixin SchedulerBinding on BindingBase {
     _postFrameCallbacks.add(callback);
   }
 
+  final List<VoidCallback> _preRenderCallbacks = <VoidCallback>[];
+
+  /// Schedule a callback to be executed just before rendering a frame.
+  ///
+  /// The provided callback run immediately after compositing
+  /// a scene [ContainerLayer.buildScene] and before rendering
+  /// it [FlutterView.render].
+  ///
+  /// The callbacks are executed in the order in which they have been
+  /// added.
+  ///
+  /// Pre-rendering callbacks cannot be unregistered. They are called exactly once.
+  ///
+  /// See also:
+  ///
+  /// * [addPostFrameCallback], which schedule a callback for the end of
+  ///   this frame.
+  void addPreRenderCallback(VoidCallback callback) {
```

Comment:
What if you have multiple views? How do I target the rendering of a specific view?

Reply:
Do you mean multiple views in this? https://github.com/flutter/engine/pull/45512
This feature is not merged into flutter so I don't have the information about the new api/changes need to be made to support it.

Reply:
That PR is just to make engine support rendering to multiple views. The framework's API already supports multiple views. https://api.flutter.dev/flutter/dart-ui/PlatformDispatcher/views.html